app/models/user.rb

```diff
@@ -4,9 +4,35 @@ class User < ApplicationRecord
   has_secure_password
   has_many :rooms, dependent: :destroy
   has_many :recordings, through: :rooms
+
+  has_one_attached :avatar
+
   validates :name, presence: true # TODO: amir - Change into full_name or seperate first and last name.
   # TODO: amir - Validate email format.
   validates :email, presence: true, uniqueness: { case_sensitive: false, scope: :provider }
   validates :provider, presence: true
   validates :password_confirmation, presence: true, on: :create
+
+  # TODO: samuel - ActiveStorage validations needs to be discussed and implemented.
+  validate :avatar_validation
+
+  after_commit :set_default_avatar, on: %i[create update]
```

Comment:
I would only do this on create

Reply:
I plan on adding a "Remove Avatar" button today.
The on update will (should) then set the default avatar as fallback of removing the avatar.

Reply:
I'm not sure I would do it like that - I'd probably manually set it to blank in the update

Reply:
What do you mean?
You would not add a "Remove Avatar" button?

Reply:
No, removing the avatar shouldn't depend on an `after_commit` to clear the avatar. I would just depend on the form data being submitted to be properly dealt with instead